landlab/components/gravel_river_transporter/gravel_river_transporter.py

```diff
@@ -0,0 +1,484 @@
+import numpy as np
+from landlab import Component, HexModelGrid
+from landlab.grid.diagonals import DiagonalsMixIn
+from scipy.sparse.linalg import spsolve
+
+_ONE_SIXTH = 1.0 / 6.0
+
+
+def make_empty_matrix_and_rhs(grid):
+    from scipy.sparse import csc_matrix
```

Comment:
Please move this import statement at the top of this file.

Reply:
The import is here instead of top-of-file because it would only be needed when using a matrix-based solver, which is optional. What is considered best practice in terms of importing libraries that might or might not be needed: do you import everything (clean but maybe expensive), or only import if/when needed (avoiding unnecessary computing and memory space but potentially messier to understand)?

Reply:
There are some cases when it is better to put the import statement within the function instead of at the top. I made this comment based on the regular coding practice to put import at the top of the file, but not aware how this library may run expensive operations (in computing or memory costs) on import. If it is more efficient to import the library at the function call, then it can be kept in the function. There are some similar comments for the import statements below. Please keep or change the code as needed. Thanks.